tests/accessor/accessor_exceptions.hpp

```diff
@@ -0,0 +1,395 @@
+          sycl_stub::target>
+struct Kernel;
+
+/**
+ * @brief Function helps to verify that constructors accessor really thrown
+ *        exceptions.
+ * @tparam AccType Current type of the accessor: generic_accessor,
+ *         local_accessor, or host_accessor
+ * @tparam DataT Current data type
+ * @tparam Dimension Dimension size
+ * @tparam Target Current target
+ * @tparam GetAccFunctorT Type of the functor that will be use to interact with
+ *         accessor
+ */
+template <accessor_tests_common::accessor_type AccType, typename DataT,
+          int Dimension, sycl_stub::target Target, typename GetAccFunctorT>
+void check_exception(GetAccFunctorT construct_acc) {
+  auto queue = util::get_cts_object::queue();
+  DataT some_data(expected_val);
+  auto r = util::get_cts_object::range<Dimension>::get(1, 1, 1);
+  {
+    sycl::buffer<DataT, Dimension> data_buf(&some_data, r);
+
+    if constexpr (AccType == accessor_type::generic_accessor) {
+      auto action = [&] {
+        queue.submit([&](sycl::handler& cgh) { construct_acc(cgh, data_buf); })
+            .wait();
+      };
+      CHECK_THROWS_MATCHES(
+          action, sycl::exception,
+          sycl_cts::util::equals_exception(sycl::errc::invalid));
+    } else if constexpr (AccType ==
+                         accessor_tests_common::accessor_type::host_accessor) {
+      auto action = [&] {
+        queue.submit([&](sycl::handler& cgh) { construct_acc(cgh, data_buf); })
+            .wait();
+      };
+      CHECK_THROWS_MATCHES(
+          action, sycl::exception,
+          sycl_cts::util::equals_exception(sycl::errc::invalid));
+    } else if constexpr (AccType ==
+                         accessor_tests_common::accessor_type::local_accessor) {
+      auto action = [&] { construct_acc(queue); };
+
+      CHECK_THROWS_MATCHES(
+          action, sycl::exception,
+          sycl_cts::util::equals_exception(sycl::errc::kernel_argument));
+    }
+  }
+}
+
+/**
+ * @brief Provides functor that lets verify that local_accessor really thrown
+ *        exception.
+ * @tparam DataT Current data type
+ * @tparam Dimension Dimension size
+ * @tparam Target Current target
+ * @tparam AccType Current type of the accessor: generic_accessor,
+ *         local_accessor, or host_accessor
+ * @param type_name Current data type string representation
+ * @param access_mode_name Current access mode string representation
+ * @param target_name Current target string representation
+ */
+template <typename DataT, int Dimension, sycl_stub::target Target,
+          accessor_tests_common::accessor_type AccType>
+void test_exception_for_local_acc(const std::string& type_name,
+                                  const std::string& access_mode_name,
+                                  const std::string& target_name) {
+  auto section_name = get_section_name<Dimension>(
+      type_name, access_mode_name, target_name,
+      "Expecting exception when attempting to interact with local_accessor on "
```

Comment:
> You probably meant [host_accessor](https://www.khronos.org/registry/SYCL/specs/sycl-2020/html/sycl-2020.html#_buffer_accessor_for_host_code).

No, I mean `local_accessor`. The `section_name` has been changed in this commit: https://github.com/KhronosGroup/SYCL-CTS/pull/329/commits/f1c24011d5ca0aebff86687b23ae0c02f7af055f

> Also I didn't find in spec exception that should be thrown, when user interacting with host_accessor on device side. Can you please give a link to this clause?

In this case I verify that that the exception with the `errc::kernel_argument` will be thrown in case a `local_accessor` is invoked via single_task